includes/class-reader-activation.php

```diff
@@ -54,15 +54,17 @@ public static function enqueue_scripts() {
 			$reader_email = \wp_get_current_user()->user_email;
 		}
 		\wp_localize_script(
-			$handle,
+			self::SCRIPT_HANDLE,
 			'newspack_reader_activation_data',
 			[
 				'auth_intention_cookie' => self::AUTH_INTENTION_COOKIE,
+				'cid_cookie'            => NEWSPACK_CLIENT_ID_COOKIE_NAME,
+				'nonce'                 => wp_create_nonce( 'wp_rest' ),
```

Comment:
@adekbadek I overlooked this line. What is the purpose of this nonce? It won't behave as expected for cached pages.

Reply:
This is necessary to [make an authenticated request from the front-end](https://developer.wordpress.org/rest-api/using-the-rest-api/authentication/#cookie-authentication). Not actually used in this PR, but [in the Google login PR](https://github.com/Automattic/newspack-plugin/pull/1781/files#diff-e9abbf90df0b524d7bfa6b07b4f954dd6092b1f061027d9a5faa6ec28f8be3a1R78-R86) (which was branched off this PR's branch, initially).

> It won't behave as expected for cached pages.

Can you elaborate on that?

Reply:
The nonce value will be cached as well, defeating the purpose of a number used once.

Do we need a nonce for `/wp-json/newspack/v1/login/google/register`? Why would that be susceptible to a CSRF attack?